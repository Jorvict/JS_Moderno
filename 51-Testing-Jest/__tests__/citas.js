import Citas from "../js/classes/Citas";


describe('Probar la clase de Citas', () => {

    const citas = new Citas();
    const id = Date.now();

    test('Agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Jorvict',
            telefono: '987456321',
            fecha: '08-06-2023',
            hora:'07:40',
            sintomas: 'Solo duerme'
        }

        citas.agregarCita(citaObj);

        // Prueba
        expect(citas).toMatchSnapshot();
    });


    test('Actualizar cita', () => {
        const citaActualizada = {
            id,
            mascota: 'Nuevo nombre',
            propietario: 'Jorvict Enrique',
            telefono: '987456321',
            fecha: '08-06-2023',
            hora:'07:40',
            sintomas: 'Solo duerme'
        }

        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();
    });


    test('Eliminar Cita', () => {
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
    });
});