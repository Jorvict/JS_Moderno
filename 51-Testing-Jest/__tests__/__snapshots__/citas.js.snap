// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de Citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "08-06-2023",
      "hora": "07:40",
      "id": 1686229281319,
      "mascota": "Nuevo nombre",
      "propietario": "Jorvict Enrique",
      "sintomas": "Solo duerme",
      "telefono": "987456321",
    },
  ],
}
`;

exports[`Probar la clase de Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "08-06-2023",
      "hora": "07:40",
      "id": 1686229281319,
      "mascota": "Hook",
      "propietario": "Jorvict",
      "sintomas": "Solo duerme",
      "telefono": "987456321",
    },
  ],
}
`;

exports[`Probar la clase de Citas Eliminar Cita 1`] = `
Citas {
  "citas": Array [],
}
`;
